src/payment/strategies/bolt/bolt-app-payment-strategy.spec.ts

```diff
@@ -0,0 +1,187 @@
+          .mockReturnValue(submitOrderAction);
+      jest.spyOn(paymentActionCreator, 'submitPayment')
+          .mockReturnValue(submitPaymentAction);
+      jest.spyOn(paymentMethodActionCreator, 'loadPaymentMethod')
+          .mockResolvedValue(store.getState());
+      jest.spyOn(store.getState().paymentMethods, 'getPaymentMethod')
+          .mockReturnValue(paymentMethodMock);
+      jest.spyOn(boltScriptLoader, 'load')
+          .mockResolvedValue(boltClient);
+
+      strategy = new BoltAppPaymentStrategy(
+        store,
+        orderActionCreator,
+        paymentActionCreator,
+        paymentMethodActionCreator,
+        boltScriptLoader
+      );
+  });
+
+  describe('#initialize', () => {
+    it('successfully initializes the bolt strategy and loads the bolt client', async () => {
+        await expect(strategy.initialize(options)).resolves.toEqual(store.getState());
+        expect(boltScriptLoader.load).toHaveBeenCalledWith('publishableKey', true);
+    });
+
+    it('fails to initialize the bolt strategy if no publishableKey is not provided', async () => {
+      paymentMethodMock.initializationData.publishableKey = null;
+      await expect(strategy.initialize(options)).rejects.toThrow(MissingDataError);
+      expect(boltScriptLoader.load).not.toHaveBeenCalled();
+    });
+  });
+
+  describe('#execute', () => {
+    let options: PaymentInitializeOptions;
+    const expectedPayment = {
+      methodId: 'bolt',
+      paymentData: {
+          nonce: 'transactionReference',
+      },
+    };
+    options = {
+      methodId: 'bolt',
+    };
+
+    it('successfully executes the bolt strategy and submits payment', async () => {
+        await strategy.initialize(options);
+        await strategy.execute(payload);
+        expect(boltClient.configure).toHaveBeenCalledWith(expect.any(Object), {}, expect.any(Object));
+        expect(paymentActionCreator.submitPayment).toHaveBeenCalledWith(expectedPayment);
+    });
+
+    it('fails to execute the bolt strategy if no payment is provided', async () => {
+      payload.payment = undefined;
+      await strategy.initialize(options);
+      await expect(strategy.execute(payload)).rejects.toThrow(InvalidArgumentError);
+      expect(boltClient.configure).not.toHaveBeenCalled();
+      expect(paymentActionCreator.submitPayment).not.toHaveBeenCalled();
+    });
+
+    it('fails to execute the bolt strategy if no client token is provided', async () => {
+      paymentMethodMock.clientToken = undefined;
+      await strategy.initialize(options);
+      await expect(strategy.execute(payload)).rejects.toThrow(MissingDataError);
+      expect(boltClient.configure).not.toHaveBeenCalled();
+      expect(paymentActionCreator.submitPayment).not.toHaveBeenCalled();
+    });
+
+    it('fails to execute the bolt strategy if the client script is not loaded', async () => {
+      jest.spyOn(boltScriptLoader, 'load')
+          .mockResolvedValue(undefined);
```

Comment:
I think this test is irrelevant, as boltScriptLoader cannot return undefined. Either throws an exception or returns the actual client.

Reply:
the purpose of this is to be able to test the case where the bolt client is not loaded, which is due to typescript trying to validate the bolt client that was declared as `private _boltClient?: BoltCheckout;` as possibly undefined, to avoid this its possible to declare it the same way as the methodId `private _boltClient!: BoltCheckout;`  but not sure if this is a good idea, as every other provider does not do this and instead validates the existence of the loaded script.